src/server/pps/server/api_server.go

```diff
@@ -492,81 +492,23 @@ func (a *apiServer) UpdateJobStateInTransaction(txnCtx *txncontext.TransactionCo
-	func() { a.Log(request, nil, nil, 0) }()
-	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
-	var result *pps.Job
-	if err := a.txnEnv.WithWriteContext(ctx, func(txnCtx *txncontext.TransactionContext) error {
-		commitInfo, err := a.env.PfsServer().InspectCommitInTransaction(txnCtx, &pfs.InspectCommitRequest{
-			Commit: request.OutputCommit,
-		})
-		if err != nil {
-			return err
-		}
-		if commitInfo.Finished != nil {
-			return pfsServer.ErrCommitFinished{Commit: commitInfo.Commit}
-		}
-		if request.Stats == nil {
-			request.Stats = &pps.ProcessStats{}
-		}
-		pipelines := a.pipelines.ReadWrite(txnCtx.SqlTx)
-		jobs := a.jobs.ReadWrite(txnCtx.SqlTx)
-		jobPtr := &pps.StoredJobInfo{
-			Job:           client.NewJob(uuid.NewWithoutDashes()),
-			OutputCommit:  request.OutputCommit,
-			Pipeline:      request.Pipeline,
-			Stats:         request.Stats,
-			Restart:       request.Restart,
-			DataProcessed: request.DataProcessed,
-			DataSkipped:   request.DataSkipped,
-			DataTotal:     request.DataTotal,
-			DataFailed:    request.DataFailed,
-			DataRecovered: request.DataRecovered,
-			StatsCommit:   request.StatsCommit,
-			Started:       request.Started,
-			Finished:      request.Finished,
-		}
-		result = jobPtr.Job
-		return ppsutil.UpdateJobState(pipelines, jobs, jobPtr, pps.JobState_JOB_STARTING, "")
-	}); err != nil {
-		return nil, err
-	}
-	return result, nil
-}
-
 // InspectJob implements the protobuf pps.InspectJob RPC
 func (a *apiServer) InspectJob(ctx context.Context, request *pps.InspectJobRequest) (response *pps.JobInfo, retErr error) {
 	func() { a.Log(request, nil, nil, 0) }()
 	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
-	if request.Job == nil && request.OutputCommit == nil {
-		return nil, errors.Errorf("must specify either a Job or an OutputCommit")
+	if request.Job == nil {
+		return nil, errors.Errorf("must specify a job")
 	}
 	jobs := a.jobs.ReadOnly(ctx)
-	if request.OutputCommit != nil {
-		if request.Job != nil {
-			return nil, errors.Errorf("can't set both Job and OutputCommit")
-		}
-		pachClient := a.env.GetPachClient(ctx)
-		ci, err := pachClient.InspectCommit(request.OutputCommit.Branch.Repo.Name, request.OutputCommit.Branch.Name, request.OutputCommit.ID)
-		if err != nil {
-			return nil, err
-		}
-		if err := a.listJob(ctx, nil, ci.Commit, nil, -1, false, "", func(ji *pps.JobInfo) error {
-			if request.Job != nil {
-				return errors.Errorf("internal error, more than 1 Job has output commit: %v (this is likely a bug)", request.OutputCommit)
-			}
-			request.Job = ji.Job
-			return nil
-		}); err != nil {
-			return nil, err
-		}
-		if request.Job == nil {
-			return nil, errors.Errorf("job with output commit %s not found", request.OutputCommit.ID)
-		}
+	// Make sure the job exists
+	// TODO: there's a race condition between this check and the watch below where
```

Comment:
We could potentially address this by having a timeout on the first event, which should return the job info as is when the watch is setup. Also, we could probably handle this by having an option on `WatchOne` that requires that the key exists.

Reply:
Yes, we've had some discussions about how to fix this, but this was an existing problem before so I consider it out of scope right now - this TODO was just to make the problem more visible.  I was thinking of maybe just having a new empty event type to indicate when the initial values are done which may be more generally useful?

Reply:
Okay, yeah we can leave this as a TODO for now, and circle back later.